### PR TITLE
Fix formula for ut_ss_benefits_credit_max variable

### LINKED ISSUE #3289: Fix formula for ut_ss_benefits_credit_max variable
Utah provides, at least as far back as 2021, a social security benefits credit as can be seen in these 2021 instructions:

Screenshot 2023-11-20 at 6 39 02 PM

However, for some unexplained reason @nikhilwoodruff coded the formula for the `ut_ss_benefits_credit_max` variable in this way:

Screenshot 2023-11-20 at 6 44 03 PM

Adding the `_2022` to the formula name seems to set the maximum value of this credit to zero in years before 2022.   But as the screenshot of the 2021 Utah income tax instructions make clear, this credit was available in 2021.  There are thousands of differences between Policyengine-US and TAXSIM35 because of this mistaken assumption that there was no social security benefits credit before 2022.

### PR DESCRIPTION
**Pull request by mzx-4real**
Fixes #3289

### Summary
📝📉🗓️

This pull request simplifies the formula for the `ut_ss_benefits_credit_max` variable in the Utah state tax system. It updates the `policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py` file and the `changelog_entry.yaml` file accordingly.

> _A pull request came with a note_
> _That it changed how a credit is wrote_
> _For `ut_ss_benefits_credit_max`_
> _It simplified the syntax_
> _And added a changelog to denote_

### Walkthrough
*  Simplify the formula for the `ut_ss_benefits_credit_max` variable by removing the year-specific suffix and the unnecessary logic for limiting the credit ([link](https://github.com/PolicyEngine/policyengine-us/pull/3420/files?diff=unified&w=0#diff-180a1aeda5fb277b7392d7045398edaa3a02ce1215fa70a64094134d7ebffdb9L13-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3420/files?diff=unified&w=0#diff-180a1aeda5fb277b7392d7045398edaa3a02ce1215fa70a64094134d7ebffdb9L40-R40))
* Add a changelog entry for the pull request, indicating a minor version bump and the change to the `ut_ss_benefits_credit_max` variable ([link](https://github.com/PolicyEngine/policyengine-us/pull/3420/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

### CODE REVIEW

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py:1]: could you add a unit test for 2021 to show that it passes

add the credit to the list of refundable/non-refundable credits

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py:1]: delete policyengine_us/parameters/gov/states/ut/tax/income/credits/ss_benefits/refundable.yaml this parameter 

and add ut_ss_benefits_credit_max to list of non-refundable credits

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: also same process needed for `ut_eitc`

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ut_eitc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ut_eitc.py:1]: can you add the ut_eitc to a list of non-refundable credits in Utah

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py:1]: can you add the credit to a list of non-refundable credits

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: patch
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+  changes:
+    changed:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixed:
```

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+      - Fix formula for ut_ss_benefits_credit_max, ut_eitc variable.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
      - Utah social security benefits credit variable.
      - Utah earned income tax credit variable.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.py:1]: need to adjust the unit tests of the variables that we changed

File: policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ut/tax/income/credits/ss_benefits_credit/ut_ss_benefits_credit_max.yaml:1]: add references and period to the non_refundable.yaml

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**